### ISSUE TITLE
Makefile dependencies of g2g

### ISSUE THREAD

**Issue by gonzalodm** [MALBECC/lio#158]
The pointxc dependencies are not set correctly. Should be:

```
cpu/iteration.o: calc_ggaCS.h calc_ggaOS.h calc_ldaCS.h
cuda/iteration.o: calc_ggaCS.h calc_ggaOS.h calc_ldaCS.h
```

Also, accumulate_point is not a cpp but a header, so there is no .o. It is not clear how to set that chain of dependencies when headers include other headers.

**Comment by ramirezfranciscof** [comment]
[This article](http://www.cplusplus.com/forum/articles/10627/) may be relevant for how to better organize header files. I'm still not sure how to solve the header dependency on other headers problem...

**Comment by ramirezfranciscof** [comment]
Once #162 is merged and this is closed, we should open a new issue to decide how to deal with headers.
